prog/dftb+/lib_reks/reksen.F90

```diff
@@ -144,13 +144,18 @@ subroutine calcSaReksEnergy(this, energy)
 
     energy%Eelec = energy%EnonSCC + energy%Escc + energy%Espin + &
         & energy%e3rd + energy%Efock
-    energy%Etotal = energy%Eelec + energy%Erep + energy%eDisp
+    energy%Etotal = energy%Eelec + energy%Erep + energy%Edisp
 
     ! Compute the total energy for SA-REKS states
     do ist = 1, this%nstates
       this%energy(ist) = sum(this%weightL(ist,:)*this%enLtot(:))
     end do
 
+!    if (abs(energy%Etotal - this%energy(this%rstate)) >= epsilon(1.0_dp)) then
+    if (abs(energy%Etotal - this%energy(this%rstate)) >= 1.0e-8_dp) then
```

Comment:
When I use epsilon(1.0_dp), this part shows an error. But Etotal and energy(rstate) show identical energy. I do not know why this condition satisfies. Is there any good option for tolerance? Currently, I use 1.0E-8.

Reply:
Could this be noise in the dispersion energy? Some of the more complex dispersion models have more numerical components to their evaluation and so will depend on several tolerances being convereged.

Reply:
Then, how to handle this error? The reason I add this error is just to check the total energy and energy contributions. This part is not mandatory. Do I remove this error?

Reply:
If the tolerance of 1E-8 is working, this can be refactored later if there is a change in the accuracies of the dispersion models. Are you seeing this problem for molecular or periodic geometries, as some of the dispersion methods sum the periodic case in real space, but should possibly be changed to used Ewald summation.

Reply:
So far I didn't test the periodic systems. For molecular geometries, tolerance of 1E-8 is working, and I have never seen this error. Then, if you agree, I will remove this line since this part is not tested for periodic systems.